mods/beds/api.lua

```diff
@@ -96,9 +115,10 @@ function beds.register_bed(name, def)
 				return false
 			end
 			node.param2 = new_param2
-			minetest.swap_node(pos, node)
-			minetest.remove_node(p)
-			minetest.set_node(newp, {name = node.name:gsub("%_bottom", "_top"), param2 = new_param2})
+			-- do not remove_node here - it will trigger destroy_bed()
```

Comment:
Turns out the order here was the problem ... lol. This works fine, nothing wrong here, though.